Resources/Prototypes/Entities/Structures/Power/Generation/generators.yml

```diff
@@ -0,0 +1,265 @@
+    nodes:
+      output:
+        !type:CableDeviceNode
+        nodeGroupID: HVPower
+  - type: PowerSupplier
+    supplyRate: 3000
+    supplyRampRate: 500
+    supplyRampTolerance: 500
+
+# Construction Frames
+
+- type: entity
+  abstract: true
+  id: BaseGeneratorWallmountFrame
+  name: wallmount generator frame
+  description: A construction frame for a wallmount generator.
+  placement:
+    mode: SnapgridCenter
+  components:
+  - type: Clickable
+  - type: InteractionOutline
+  - type: Physics
+  - type: Fixtures
+    fixtures:
+      - shape:
+          !type:PhysShapeAabb
+            bounds: "-0.25,-0.25,0.3,0.25"
+        layer: [ Passable ]
+  - type: Transform
+    anchored: true
+  - type: Sprite
+    drawdepth: WallMountedItems
+    netsync: false
+    sprite: Structures/Power/Generation/wallmount_generator.rsi
+    layers:
+    - state: panel
+  - type: Construction
+    graph: WallmountGenerator
+    node: frame
+
+# Generators in use
+
+- type: entity
+  parent: BaseGenerator
+  id: GeneratorBasic
+  suffix: Basic, 3kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 3000
+    
+- type: entity
+  parent: BaseGenerator
+  id: GeneratorPlasma
+  suffix: Plasma, 5kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 5000 # 260 sec / sheet
+  - type: Sprite
+    sprite: Structures/Power/Generation/portable_generator.rsi
+    state: portgen0_1
+  - type: Construction
+    graph: Machine
+    node: machine
+  - type: Wires
+    BoardName: "GeneratorPlasma"
+    LayoutId: GeneratorPlasma
+  - type: Machine
+    board: GeneratorPlasmaMachineCircuitboard
+    
+- type: entity
+  parent: BaseGenerator
+  id: GeneratorUranium
+  suffix: Uranium, 15kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 15000 # 85 sec / sheet
+  - type: Sprite
+    sprite: Structures/Power/Generation/portable_generator.rsi
+    state: portgen1_1
+  - type: Construction
+    graph: Machine
+    node: machine
+  - type: Wires
+    BoardName: "GeneratorUranium"
+    LayoutId: GeneratorUranium
+  - type: Machine
+    board: GeneratorUraniumMachineCircuitboard
+
+
+- type: entity
+  parent: BaseGeneratorWallmount
+  id: GeneratorWallmountBasic
+  suffix: Basic, 3kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 3000
+  - type: Construction
+    graph: WallmountGenerator
+    node: generator
+
+- type: entity
+  parent: BaseGeneratorWallmount
+  id: GeneratorWallmountAPU
+  name: shuttle APU
+  description: An auxiliary power unit for a shuttle - 6kW.
+  suffix: APU, 6kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 6000
+  - type: Construction
+    graph: WallmountGenerator
+    node: APU  
+
+# RTG - no fuel requirement
+
+- type: entity
+  parent: BaseGenerator
+  id: GeneratorRTG
+  name: RTG
+  description: A Radioisotope Thermoelectric Generator for long term power.
+  suffix: 20kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 20000 # supposed to be low power at no fuel cost to power derelicts
```

Comment:
low power? realistically they're going to unanchor it and bring it to the station which seems a little overpowered.

Reply:
Supposed to be. But yes taking it off derelicts is half intended. It'll be for the big big salvages since small ones can just run off an APC or one of these new substations.